test/OffsetHelper.test.ts

```diff
@@ -121,7 +125,102 @@ describe("OffsetHelper", function () {
+      ](fromToken.address, fromAmount, toToken.address);
+      // sanity check, should easily be > 1 tonne for all provided inputs
+      expect(expOffset).to.be.greaterThan(ONE_ETHER);
+
+      await fromToken.approve(offsetHelper.address, fromAmount);
+
+      const supplyBefore = await toToken.totalSupply();
+      const autoOffsetUsingToken =
+        offsetHelper["autoOffsetUsingToken(address,uint256,address)"];
+      await expect(
+        autoOffsetUsingToken(fromToken.address, fromAmount, toToken.address)
+      )
+        .to.emit(offsetHelper, "Redeemed")
+        .withArgs(
+          addr2.address,
+          toToken.address,
+          anyValue,
+          (amounts: BigNumber[]) => {
+            return expOffset == sumBN(amounts);
+          }
+        )
+        .and.to.changeTokenBalance(
+          fromToken,
+          addr2.address,
+          fromAmount.mul(-1)
+        );
+
+      const supplyAfter = await toToken.totalSupply();
+      expect(supplyBefore.sub(supplyAfter)).to.equal(expOffset);
+    }
+
+    TOKEN_POOLS.forEach((pool) => {
+      it(`should retire 1 WETH for ${pool.name} redemption`, async function () {
+        await retireFixedInToken(weth, ONE_ETHER, pool.token());
+      });
+
+      it(`should retire 100 USDC for ${pool.name} redemption`, async function () {
+        await retireFixedInToken(usdc, parseUSDC("100"), pool.token());
+      });
+
+      it(`should retire 20 WMATIC for ${pool.name} redemption`, async function () {
+        await retireFixedInToken(wmatic, parseEther("20"), pool.token());
+      });
+    });
+  });
+
+  describe("#autoOffsetUsingETH(address)[fixed-in]", function () {
+    async function retireFixedInETH(
+      fromAmount: BigNumber,
+      toToken: IToucanPoolToken
+    ) {
+      const expOffset = await offsetHelper[
+        "calculateExpectedPoolTokenAmount(uint256,address)"
+      ](fromAmount, toToken.address);
+      // sanity check, should easily be > 1 tonne for all provided inputs
+      expect(expOffset).to.be.greaterThan(ONE_ETHER);
+
+      const supplyBefore = await toToken.totalSupply();
+      const autoOffsetUsingETH = offsetHelper["autoOffsetUsingETH(address)"];
+      await expect(autoOffsetUsingETH(toToken.address, { value: fromAmount }))
+        .to.emit(offsetHelper, "Redeemed")
+        .withArgs(
+          addr2.address,
+          toToken.address,
+          anyValue,
+          (amounts: BigNumber[]) => {
+            return expOffset == sumBN(amounts);
+          }
+        )
+        .and.to.changeEtherBalance(addr2.address, fromAmount.mul(-1));
+
+      const supplyAfter = await toToken.totalSupply();
+      expect(supplyBefore.sub(supplyAfter)).to.equal(expOffset);
+    }
+
+    TOKEN_POOLS.forEach((pool) => {
+      it(`should retire 20 MATIC for ${pool.name} redemption`, async function () {
+        await retireFixedInETH(parseEther("20"), pool.token());
+      });
+    });
+  });
+
+  describe("#autoOffset()[var-in]", function () {
```

Comment:
It took me a while to figure out what `var-in` means. I think `fixed-out` would be easier to understand especially since it's in line with Uni/Sushi contracts (i.e.: `swapExactTokensForTokens` vs `swapTokensForExactTokens`)

Reply:
Agree, renamed them to `exact-in` and `exact-out` to be more in line with the Uniswap nomenclature.